problem_builder/v1/upgrade.py

```diff
@@ -148,15 +144,46 @@ def parse_xml_for_HtmlDescriptor(cls, node, runtime, keys, id_generator):
     blocks_found = []
 
     def find_mentoring_blocks(block):
-        if isinstance(block, MentoringBlock) and block.xml_content:  # If it's a v1 block it will have xml_content
+        """
+        Find mentoring and recently-upgraded blocks. We check the recently upgraded ones
+        in case an error happened and we're re-running the upgrade.
+        """
+        # If it's a v1 block or a recently upgraded block it will have xml_content:
+        if isinstance(block, (MentoringBlock, NewMentoringBlock)) and block.xml_content:
             blocks_found.append(block.scope_ids.usage_id)
         elif block.has_children:
             for child_id in block.children:
                 find_mentoring_blocks(block.runtime.get_block(child_id))
     find_mentoring_blocks(course)
+
     total = len(blocks_found)
     print(" ➔ Found {} mentoring blocks".format(total))
 
+    print(" ➔ Doing a quick sanity check of the url_names")
+    url_names = set()
+    stop = False
+    for block_id in blocks_found:
+        block = course.runtime.get_block(block_id)
+        if block.url_name in url_names:
+            print(u" ➔ Mentoring block {} conflicts with another block using the same url_name".format(block.url_name))
+            print(u'   (display_name: "{}", parent {}: "{}")'.format(
+                block.display_name, block.parent, block.get_parent().display_name
+            ))
+            stop = True
+            continue
+        if block.url_name != unicode(block_id.block_id):
+            print(u" ➔ Mentoring block {} has two different url_names assigned. Real url_name is {}".format(
+                unicode(block_id.block_id)
+            ))
+            print(u'   (display_name: "{}", parent: "{}")'.format(block.display_name, block.get_parent().display_name))
+            print(u"   (to fix, edit the XML and set the url_name to the real value given above).")
+            stop = True
+        url_names.add(block.url_name)
+
+    if stop:
+        print(" ➔ Exiting due to errors preventing the upgrade.")
+        sys.exit(1)
```

Comment:
@bradenmacdonald nit: might make sense to merge the two lines into one:

    sys.exit(" ➔ Exiting due to errors preventing the upgrade.")

According to [docs](https://docs.python.org/2/library/sys.html#sys.exit) it would yeild the same exit code of 1.